src/quacc/recipes/newtonnet/ts.py

```diff
@@ -264,6 +267,246 @@ def quasi_irc_job(
     return relax_summary
 
 
+@job
+@requires(
+    has_newtonnet, "NewtonNet must be installed. Refer to the quacc documentation."
+)
+@requires(
+    has_geodesic_interpolate,
+    "geodesic-interpolate must be installed. Refer to the quacc documentation.",
+)
+def neb_job(
+    reactant_atoms: Atoms,
+    product_atoms: Atoms,
+    relax_job_kwargs: dict[str, Any] | None = None,
+    calc_kwargs: dict[str, Any] | None = None,
+    geodesic_interpolate_kwargs: dict[str, Any] | None = None,
+    neb_kwargs: dict[str, Any] | None = None,
+) -> dict:
```

Comment:
Please use a more descriptive type hint than `dict`. In this case, you want to define a custom `TypedDict` class like

```python
NebSchema(TypedDict):
    relax_reactant: OptSchema
    relax_product: OptSchema
    ...
```

and then have this be the return type.